backend/init_app.py

```diff
@@ -29,6 +47,12 @@
 api.add_resource(Categories, '/api/category/<category_id>')
 api.add_resource(SubcategoryList, '/api/subcategory')
 api.add_resource(Subcategories, '/api/subcategory/<subcategory_id>')
```

Comment:
Sorry, this does not belong to this PR but I haven't seen it any earlier. I guess from name that subcategory is always under some category, so le'ts mirror it in URL schema! :)

```suggestion
api.add_resource(Subcategories, '/api/categories/<category_id>/subcategory/<subcategory_id>')
```
BTW Resources URL have to be in plural form.
http://www.andrewhavens.com/posts/20/beginners-guide-to-creating-a-rest-api/ (I have to read it too... :D)